xs/sources/xsSnapshot.c

```diff
@@ -1996,12 +2093,13 @@ int fxWriteSnapshot(txMachine* the, txSnapshot* snapshot)
 		mxThrowIf((*snapshot->write)(snapshot->stream, "SIGN", 4));
 		mxThrowIf((*snapshot->write)(snapshot->stream, snapshot->signature, snapshot->signatureLength));
 
-		size = 8 + sizeof(txCreation) + sizeof(txID);
+		size = 8 + sizeof(txCreation) + sizeof(txID) + sizeof(txInteger);
 		size = htonl(size);
 		mxThrowIf((*snapshot->write)(snapshot->stream, &size, 4));
 		mxThrowIf((*snapshot->write)(snapshot->stream, "CREA", 4));
 		mxThrowIf((*snapshot->write)(snapshot->stream, &creation, sizeof(txCreation)));
 		mxThrowIf((*snapshot->write)(snapshot->stream, &(the->profileID), sizeof(txID)));
+		mxThrowIf((*snapshot->write)(snapshot->stream, &(the->tag), sizeof(txInteger)));
```

Comment:
note to self: this is a snapshot compatibility break: the new snapshot needs to include space for both `incrementalKeyCount` and `the->tag` (which is really the number of keys currently allocated).